diff --git a/CMakeLists.txt b/CMakeLists.txt
index f77a739..5bf22e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -281,6 +281,7 @@ target_link_libraries(${PDAL_BASE_LIB_NAME}
         ${PDAL_ARBITER_LIB_NAME}
         ${PDAL_KAZHDAN_LIB_NAME}
         ${PDAL_JSONCPP_LIB_NAME}
+	${CONAN_LIBS_OPENSSL}
     INTERFACE
         ${PDAL_LIBDIR}
 )
diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index 950972c..35b1a11 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -24,6 +24,7 @@ target_link_libraries(${PDAL_APP}
         ${PDAL_JSONCPP_LIB_NAME}
         ${GDAL_LIBRARY}
         ${WINSOCK_LIBRARY}
+	${CURL_LIBRARIES}
 )
 target_include_directories(${PDAL_APP} PRIVATE
     ${PDAL_JSONCPP_INCLUDE_DIR}
diff --git a/cmake/curl.cmake b/cmake/curl.cmake
index 0d24b3f..fa6f92c 100644
--- a/cmake/curl.cmake
+++ b/cmake/curl.cmake
@@ -12,5 +12,6 @@ if (CURL_FOUND)
     else()
         add_definitions("-DUNIX")
     endif()
+    set(CURL_LIBRARIES ${CURL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
